pymc_marketing/model_config.py

```diff
@@ -0,0 +1,484 @@
+            param,
+            parameter_config,
+            dim_handler=dim_handler,
+        )
+
+    if isinstance(parameter_config, int | float | np.ndarray | pt.TensorVariable):
+        return parameter_config
+
+    raise ModelConfigError(f"{name}_{param}")
+
+
+def handle_parameter_distributions(
+    name: str,
+    param_distributions: dict[str, dict[str, Any]],
+    dim_handler: DimHandler,
+) -> dict[str, Any]:
+    return {
+        param: handle_parameter_configurations(
+            name,
+            param,
+            parameter_config,
+            dim_handler=dim_handler,
+        )
+        for param, parameter_config in param_distributions.items()
+    }
+
+
+def create_distribution(
+    name: str,
+    distribution_name: str,
+    distribution_kwargs: dict[str, Any],
+    **kwargs,
+) -> pt.TensorVariable:
+    """Create a PyMC distribution with the specified parameters.
+
+    Parameters
+    ----------
+    name : str
+        Name of the variable.
+    distribution_name : str
+        Name of the PyMC distribution.
+    distribution_kwargs : Dict
+        parameters for the distribution including any nested distributions
+    **kwargs
+        Additional keyword arguments for the distribution.
+
+    """
+    dim_handler = create_dim_handler(kwargs.get("dims"))
+    parameter_distributions = handle_parameter_distributions(
+        name, distribution_kwargs, dim_handler=dim_handler
+    )
+    distribution = get_distribution(name=distribution_name)
+    return distribution(name, **parameter_distributions, **kwargs)
+
+
+def create_distribution_from_config(name: str, config) -> pt.TensorVariable:
+    """Wrapper around create_distribution that uses a configuration dictionary.
+
+    Parameters
+    ----------
+    name : str
+        Name of the variable.
+    config : Dict
+        A configuration with the name mapping to parameter configuration.
+
+    Examples
+    --------
+
+    .. code-block:: python
+
+        import pymc as pm
+
+        from pymc_marketing.model_config import create_distribution_from_config
+
+        distribution = {
+            "dist": "Normal",
+            "kwargs": {"mu": 0, "sigma": 1},
+        }
+        config = {
+            "alpha": distribution,
+            "beta": distribution,
+        }
+
+        with pm.Model():
+            alpha = create_distribution_from_config("alpha", config)
+
+    """
+    parameter_config = config[name]
+    try:
+        dist_name = parameter_config["dist"]
+        dist_kwargs = parameter_config["kwargs"]
+    except KeyError:
+        raise ModelConfigError(name)
+
+    return create_distribution(
+        name,
+        dist_name,
+        dist_kwargs,
+        dims=parameter_config.get("dims"),
+    )
+
+
+LIKELIHOOD_DISTRIBUTIONS: set[str] = {
+    "Normal",
+    "Beta",
+    "StudentT",
+    "Laplace",
+    "Logistic",
+    "LogNormal",
+    "Wald",
+    "TruncatedNormal",
+    "Gamma",
+    "AsymmetricLaplace",
+    "VonMises",
```

Comment:
Can we sort this alphabetically (ask ChatGPT) :) ?

Reply:
vim has built-in sort function 😉